### DISCUSSION TITLE
How to extend the expiration date?

### DISCUSSION THREAD

**Discussion by BrightXiaoHan** [dashingsoft/pyarmor#1319] [Q&A] [answered]
I sold a software to a user, for example, with a one-year expiration date. If the user later pays to extend the usage time for another year, do I have to repackage the application, or is there a method to directly update the license for the user without the need for repackaging?

**Comment by jondy** [comment, marked as answer]
Using outer runtime key, but you need sent new key file to your customer by someway, pyarmor doesn't provide method to update it automatically, please check documentation for the usage of outer runtime key.
